gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -113,6 +119,25 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     viewModel.initialize(EditObservationFragmentArgs.fromBundle(getArguments()));
   }
 
+  private void fetchResponsesAndErrors() {
+    for (AbstractFieldViewModel fieldViewModel : fieldViewModels) {
+      String fieldId = fieldViewModel.getField().getId();
+      viewModel.onResponseChanged(fieldId, fieldViewModel.getResponse());
```

Comment:
It appears we're mixing sync and async here when we could return the responses as a stream to the fragment -> vm. Please cmiiw.